game-core/src/main/java/games/strategy/triplea/image/TerritoryEffectImageFactory.java

```diff
@@ -0,0 +1,63 @@
+package games.strategy.triplea.image;
+
+import java.awt.Image;
+import java.awt.Toolkit;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.swing.ImageIcon;
+
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.triplea.ResourceLoader;
+import games.strategy.ui.Util;
+
+/**
+ * Used to manage territory effect images.
+ */
+public class TerritoryEffectImageFactory {
+
+  private static final String FILE_NAME_BASE = "territoryEffects/";
+  private final Map<String, ImageIcon> icons = new HashMap<>();
+  private ResourceLoader resourceLoader;
+
+  public TerritoryEffectImageFactory() {}
+
+  public void setResourceLoader(final ResourceLoader loader) {
+    resourceLoader = loader;
+    clearImageCache();
+  }
+
+  private void clearImageCache() {
+    icons.clear();
+  }
+
+  private Image getBaseImage(final String baseImageName) {
```

Comment:
I get an IDE copy/paste warning for these methods:
![copy_pste](https://user-images.githubusercontent.com/12397753/36626360-30ef0dee-18e6-11e8-9741-09f728eca517.png)


`ResourceImageFactory` and `TerritoryEffectImageFactory` share 25 lines in exact common, requesting changes to fix this at least.

Reply:
Designing for inheritance is surprisingly difficult. I'd vote for utility methods between the two. Strategy object is a third potential option, but utility method at this point is likely cleaner/easier.

Looking at this some more, the new `getIcon` is only called in one place. We can simplify a bit as we know the value of the boolean parameter.

Reply:
Went ahead and created AbstractImageFactory to hold common methods across Resource and TerritoryEffect image factories.